### ISSUE TITLE
Error when running evaluation

### ISSUE THREAD

**Issue by MengyuanChen21** [gicheonkang/gst-visdial#5]
Hello author, first of all, thank you for your excellent work!

I ran into an error when performing evaluation with the codes, the command and the checkpoint provided. 

The command is:

python evaluate_gen.py \
  -mode vd_eval_val \
  -start_path checkpoints/student_v1.0_iter3.ckpt \
  -save_path results \
  -save_name gen.txt \
  -gpu_ids 1 2 5 6

The error report is:

  File "evaluate_gen.py", line 188, in <module>
    ranks_json = evaluate(model, dataloader, params, eval_batch_size, mode=mode)
  File "evaluate_gen.py", line 118, in evaluate
    ndcg.observe(output, gt_relevance)
  File "/home/mmc_mychen/gst-visdial/utils/visdial_metrics.py", line 163, in observe
    dcg = self._dcg(
  File "/home/mmc_mychen/gst-visdial/utils/visdial_metrics.py", line 177, in _dcg
    sorted_relevance = relevance[rankings].cpu().float()
RuntimeError: indices should be either on cpu or on the same device as the indexed tensor (cpu)

It seems that something is placed on the wrong device. I am not familiar with this project right now. Could u please help me solve it?

**Comment by gicheonkang** [comment]
Hello, thank you for your interests
I think the problem comes from the version of pytorch.

I've installed the PyTorch version 1.7.0 or 1.7.1 and never seen the error you showed.
I think the immediate solution is to change the version of pytorch.
If you want to fix the problem, can you tell me the version of pytorch you use?

Thanks!

**Comment by MengyuanChen21** [comment]
Thanks a lot! I have solved the problem by changing the version of pytorch.

PS. I would like to remind you that the environment setup part in the README.md file does not work properly (at least for me). There seems to be a version incompatibility issue with the environment configuration file.

Here is a simple script which I have used to successfully establish the environment:

conda create -n gst python=3.8 -y
conda activate gst
pip install torch==1.7.1+cu110 torchvision==0.8.2+cu110 torchaudio==0.7.2 -f https://download.pytorch.org/whl/torch_stable.html
pip install transformers pytorch-transformers pytorch_pretained_bert lmdb

Thanks again for your help!
